File: core/src/dc/targetman/gamelogic/StickActions.java
package dc.targetman.gamelogic;

import dc.targetman.epf.parts.MovementPart;
import dc.targetman.epf.parts.WeaponPart;
import dclib.epf.Entity;

public final class StickActions {

	private StickActions() {
	}

	public static final void move(final Entity entity, final float direction) {
		entity.get(MovementPart.class).setDirection(direction);
	}

	public static final void jump(final Entity entity) {
		entity.get(MovementPart.class).setJumping(true);
	}

	public static final void aim(final Entity entity, final float direction) {
		entity.get(WeaponPart.class).setAimDirection(direction);
	}

	public static final void trigger(final Entity entity) {
		entity.get(WeaponPart.class).setTriggered(true);
	}

}

File: core/src/main/dc/targetman/character/StickActions.kt
package dc.targetman.character

import dc.targetman.epf.parts.MovementPart
import dc.targetman.epf.parts.WeaponPart
import dc.targetman.mechanics.Direction
import dclib.epf.Entity

object StickActions {
    fun move(entity: Entity, direction: Direction) {
        entity[MovementPart::class].direction = direction
    }

    fun jump(entity: Entity) {
        entity[MovementPart::class].tryJumping = true
    }

    fun aim(entity: Entity, direction: Int) {
        entity[WeaponPart::class].aimDirection = direction
    }

    fun trigger(entity: Entity) {
        entity[WeaponPart::class].setTriggered(true)
    }
}
